libminifi/src/utils/HTTPClient.cpp

```diff
@@ -15,8 +15,33 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include "utils/HTTPClient.h"
+#include <algorithm>
 #include <string>
+
+#include "utils/HTTPClient.h"
+#include "utils/StringUtils.h"
+
+namespace {
+
+constexpr const char* HTTP = "http://";
+constexpr const char* HTTPS = "https://";
+
+utils::optional<int> parsePortNumber(const std::string& port_string) {
+  try {
+    size_t pos;
+    int port = std::stoi(port_string, &pos);
+    if (pos == port_string.size()) {
+      return port;
+    }
+  } catch (const std::invalid_argument&) {
+  } catch (const std::out_of_range&) {
+  }
+
+  return {};
+}
```

Comment:
Just an opinion, but these might fit better as static members of the URL class.

Reply:
I think this way is better for encapsulation, as `parsePortNumber()` does not need access to any members of the `URL` class.